src/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -742,16 +789,14 @@ internal static bool IsLWS(char ch)
         }
 
         //Only consider ASCII characters
-        internal static bool IsAsciiLetter(char character)
-        {
-            return (character >= 'a' && character <= 'z') ||
-                   (character >= 'A' && character <= 'Z');
-        }
-
-        internal static bool IsAsciiLetterOrDigit(char character)
-        {
-            return IsAsciiLetter(character) || (character >= '0' && character <= '9');
-        }
+        internal static bool IsAsciiLetter(char character) =>
+            ((uint)(character - 'a') <= 'z' - 'a') ||
+            ((uint)(character - 'A') <= 'Z' - 'A');
+
+        internal static bool IsAsciiLetterOrDigit(char character) =>
+            ((uint)(character - 'a') <= 'z' - 'a') ||
+            ((uint)(character - 'A') <= 'Z' - 'A') ||
+            ((uint)(character - '0') <= '9' - '0');
```

Comment:
unrelated: I see hundreds of such hand-made checks for isAscii (with `uint` casts) here and there in the BCL, I think it makes sense to expose it in `char` (e.g. `public bool IsInRange(char start, char end)`)

Reply:
Is there an issue tracking this?  If not, can you open one?  Thanks.